include/yamail/resource_pool/async/detail/pool_impl.hpp

```diff
@@ -301,53 +295,20 @@ std::size_t pool_impl<V, M, I, Q>::assert_capacity(std::size_t value) {
 }
 
 template <class V, class M, class I, class Q>
-template <class Handler>
-bool pool_impl<V, M, I, Q>::alloc_resource(io_context_t& io_context, unique_lock& lock, Handler&& handler) {
-    while (!_available.empty()) {
-        const list_iterator res_it = _available.begin();
-        if (res_it->drop_time <= time_traits::now()) {
-            res_it->value.reset();
-            _wasted.splice(_wasted.end(), _available, res_it);
-            continue;
-        }
-        _used.splice(_used.end(), _available, res_it);
-        lock.unlock();
-        asio::post(io_context,
-            make_on_list_iterator_handler(
-                boost::system::error_code(),
-                res_it,
-                std::forward<Handler>(handler)
-            ));
-        return true;
-    }
-    return false;
-}
-
-template <class V, class M, class I, class Q>
-template <class Handler>
-bool pool_impl<V, M, I, Q>::reserve_resource(io_context_t& io_context, unique_lock& lock, Handler&& handler) {
-    const list_iterator res_it = _wasted.begin();
-    _used.splice(_used.end(), _wasted, res_it);
+void pool_impl<V, M, I, Q>::perform_one_request(unique_lock& lock) {
+    io_context_t* io_context;
+    queued_handler handler;
+    if (!_callbacks->pop(io_context, handler))
+        return;
+    const auto cell = storage_.lease();
+    assert(cell);
```

Comment:
Is it really needed? Could it be covered by the unit tests?

Reply:
> Is it really needed?

To make it more clear why there is no check for optional state.

> Could it be covered by the unit tests?

It's covered by the unit tests.

Reply:
> To make it more clear why there is no check for optional state.

Ok.